### PR TITLE
[ANCHOR-687] Upgrade to JDK-17

### PR DESCRIPTION
**Pull request by lijamie98**
### Description

- Upgrade JDK from 11 to 17.
- Fix Jwt Json serialization

### Context

- JDK 11 is expected to sunset in 10/2024. 

### Testing

- `./gradlew test`

### Documentation

Update `How to Contribute` doc.

### Known limitations

N/A

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-ARG BASE_IMAGE=gradle:7.6.4-jdk11-alpine
+ARG BASE_IMAGE=gradle:7.6.4-jdk17-alpine
```
Comment (Ifropc) [inline Dockerfile:1]: We should upgrade gradle to latest in 3.0 if we haven't already

File: core/src/main/java/org/stellar/anchor/util/Log.java [hunk 2]
```diff
@@ -81,7 +83,19 @@ public static void error(final String msg) {
    * @param detail The additional object to be logged.
    */
   public static void error(final String message, final Object detail) {
-    logMessageWithJson(message, detail, getLogger()::error);
+    if (detail instanceof Exception) {
+      Exception ex = (Exception) detail;
+
+      logMessageWithJson(
```
Comment (Ifropc) [inline core/src/main/java/org/stellar/anchor/util/Log.java:89]: Why is this required now? Does the message get collapsed?

**Review by Ifropc — APPROVED** [review]
LGTM

File: core/src/main/java/org/stellar/anchor/util/Log.java (hunk 2 above)
Comment (lijamie98) [reply core/src/main/java/org/stellar/anchor/util/Log.java:89]: This is because `logMessageWithJson(message, detail, getLogger()::error);` failed serializing the `message` field if `detail` is an exception. 

That's the reason that we have expand it manually.

File: Dockerfile (hunk 1 above)
Comment (lijamie98) [reply Dockerfile:1]: I will try to make least impact for `2.x`. We can do major upgrades in v3.

Comment (Ifropc) [reply Dockerfile:1]: Agree!

**Review by philipliu — APPROVED** [review]
